lib/ConstPlugin.js

```diff
@@ -534,6 +502,56 @@ class ConstPlugin {
 			}
 		);
 	}
+
+	/**
+	 * Eliminate a dead statement.
+	 * @param {JavascriptParser} parser the parser
+	 * @param {Statement} statement the statement to remove
+	 * @returns {void}
+	 */
+	eliminateDeadStatement(parser, statement) {
+		// Before removing the dead branch, the hoisted declarations
+		// must be collected.
+		//
+		// Given the following code:
+		//
+		//     if (true) f() else g()
+		//     if (false) {
+		//       function f() {}
+		//       const g = function g() {}
+		//       if (someTest) {
+		//         let a = 1
+		//         var x, {y, z} = obj
+		//       }
+		//     } else {
+		//       …
+		//     }
+		//
+		// the generated code is:
+		//
+		//     if (true) f() else {}
+		//     if (false) {
+		//       var f, x, y, z;   (in loose mode)
+		//       var x, y, z;      (in strict mode)
+		//     } else {
+		//       …
+		//     }
+		//
+		// NOTE: When code runs in strict mode, `var` declarations
+		// are hoisted but `function` declarations don't.
+		//
+		const declarations = parser.scope.isStrict
+			? getHoistedDeclarations(statement, false)
+			: getHoistedDeclarations(statement, true);
+		const replacement =
+			declarations.length > 0 ? `{ var ${declarations.join(", ")}; }` : "{}";
+		const dep = new ConstDependency(
+			replacement,
```

Comment:
Let's add a comment `// removed by dead control flow\n${replacement}`  (maybe better) for debug reasons